### PR TITLE
fix: debugging star exit

### PR DESCRIPTION
**Pull request by harrysolovay**

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -75,7 +75,13 @@ async function after() {
...
     const status = await process.status()
-    self.addEventListener("unload", () => Deno.exit(status.code))
+    self.addEventListener("unload", () => {
+      console.log("exiting fine")
+      console.log(Deno.resources())
+      return Deno.exit(status.code)
+    })
+    console.log(Deno.resources())
     controller.abort()
+    setTimeout(() => console.log(Deno.resources()), 100)
```
Comment (harrysolovay) [inline main.ts:85]: Sneaky ;)

**Comment by kratico** [comment]
@harrysolovay @tjjfvi this is interesting.
when `stdout/stderr` are `piped`, to close the file descriptors we either need to consume them or close them explicitly.


```
> const p = Deno.run({cmd: ["echo", "1"], stdout: "piped", stderr: "piped"})
undefined
> Deno.resources()
{
  "0": "stdin",
  "1": "stdout",
  "2": "stderr",
  "3": "childStdout",
  "4": "childStderr",
  "5": "child"
}
> p.close()
undefined
> Deno.resources()
{ "0": "stdin", "1": "stdout", "2": "stderr", "3": "childStdout", "4": "childStderr" }
> await Promise.all([p.output(), p.stderrOutput()])
[ Uint8Array(2) [ 49, 10 ], Uint8Array(0) [] ]
> Deno.resources()
{ "0": "stdin", "1": "stdout", "2": "stderr" }
```

**Comment by kratico** [comment]
also `await p.status()` does not release the subprocess resources

```
> const p = Deno.run({cmd: ["echo", "1"]})
undefined
> 1
> await p.status()
{ success: true, code: 0 }
> Deno.resources()
{ "0": "stdin", "1": "stdout", "2": "stderr", "3": "child" }
```

**Comment by harrysolovay** [comment]
Strange then that the process exits gracefully locally. Perhaps we should create a util that accepts a process and then closes+kills and then use that util everywhere we currently `.close()`?

**Comment by harrysolovay** [comment]
Closing in favor of #648

**Pull request closed without merging**